### ISSUE TITLE
[TextField] Borders fail WCAG SC 1.4.11 Non-text Contrast

### ISSUE THREAD

**Issue by aardrian** [Shopify/polaris-react-archive#7838]
## Issue summary
The Shopify Polaris [Text Field component](https://polaris.shopify.com/components/text-field) has borders of two different colors. The _darker_ of the two (`rgb(174, 180, 185)`) has a 2.1:1 contrast ratio with the white background of the field and 2:1 with the gray of the page background.

To meet WCAG Success Criterion [1.4.11 Non-text Contrast (Level AA)](https://www.w3.org/WAI/WCAG21/Understanding/non-text-contrast.html) it must have a 3:1 contrast ratio at the minimum:

> The visual [presentation](https://www.w3.org/WAI/WCAG21/Understanding/non-text-contrast.html#dfn-presentation) of the following have a [contrast ratio](https://www.w3.org/WAI/WCAG21/Understanding/non-text-contrast.html#dfn-contrast-ratio) of at least 3:1 against adjacent color(s):
> *User Interface Components*
>    Visual information required to identify [user interface components](https://www.w3.org/WAI/WCAG21/Understanding/non-text-contrast.html#dfn-user-interface-component) and [states](https://www.w3.org/WAI/WCAG21/Understanding/non-text-contrast.html#dfn-state), except for inactive components or where the appearance of the component is determined by the user agent and not modified by the author;

Since this is not the default control (your team has added styles) and is not disabled (or at least not always) and the layout cannot always be relied upon as a sufficient to identify the fields, this component is on the hook for this SC.

## Expected behavior

All four borders (not just the top border) should meet a _minimum_ 3:1 contrast ratio with the page background, or, if the color is so light as to be subsumed into the field, the field itself must meet the contrast ratio (this is explained in the Understanding document I linked).

## Actual behavior
<img width="608" alt="Current Shopify text box has a top border of rgb(174,180,185), which gives it a 2.1:1 contrast ratio with white and 2:1 contrast with the gray background of its page." src="https://user-images.githubusercontent.com/1376607/205456084-98016ce2-4775-4ef9-8b16-dbcfbc944a01.png">

The current Shopify text box has a top border of `rgb(174,180,185)`, which gives it 2.1:1 contrast ratio with the white field and 2:1 contrast with the gray background of its page.

## Steps to reproduce the problem

1. Go to https://polaris.shopify.com/components/text-field
2. Run any automated accessibility checker, or...
3. Use the browser dev tools to inspect the colors.
4. Observe the failing contrast.

## Additional Information

I can see an intern tried to fix this in 2020 in PR [#3658 [TextField] Fix border-top color to meet 3:1 contrast ratio](https://github.com/Shopify/polaris/pull/3658). A Shopify staffer even suggested the intern [push the contrast value _down_ to 3:1](https://github.com/Shopify/polaris/pull/3658#pullrequestreview-538655001) from the proposed colors that would set it at 3.04:1. Literally trying to make it meet WCAG at the barest possible level. Remember that WCAG itself is literally the baseline of digital accessibility, the barest of bare minimums organizations must follow according to many laws (or case law).

It is unclear why this PR was never merged and why the branch was deleted, but Shopify clearly knew about this WCAG violation in late 2020.

It is also unclear why only the top border is darker, though it feels like this is an effort to push the line on what the bare minimum of WCAG conformance means. Regardless of the accessibility issue, others have noticed this inconsistency, such as issue [#3396 [new DL] Textfield has a darker top border](https://github.com/Shopify/polaris/issues/3396) where the response was [that this is intentional](https://github.com/Shopify/polaris/issues/3396#issuecomment-704925652).

The Polaris [Foundations page talks about accessibility](https://polaris.shopify.com/foundations/accessibility) extensively:

> Making commerce better for everyone means caring deeply about making quality products. A quality product should have a fantastic user experience (UX).

>It’s important that Shopify products—and [Partner](https://www.shopify.ca/partners) products—are usable and useful to everyone.

> Everyone is a pretty big group. It includes our merchants, their customers, our developer partners, our employees, and the greater tech community at large. That also includes all members of our community who have disabilities.

Allowing a documented WCAG failure to exist for 2 years and attempting to push the WCAG conformance down to the bare minimum do not feel like the actions of an organization that cares deeply about everyone. Users, customers, partners, and especially the intern who tried to do the right (and legal) thing and was instead shown how to do less than that.

## Specifications

- Are you using the React components? (Y/N): N
- Polaris version number: current
- Browser: Firefox, Chrome, Safari
- Device: Laptop, mobile
- Operating System: Windows 11, macOS 12.6.1, Android 13, iOS 16.1, iPadOS 16.1

**Comment by heyjoethomas** [comment]
Hey @aardrian :wave: Thank you so much for flagging this.
After reading through https://github.com/Shopify/polaris/pull/3658, it looks like the color change was being hard-coded in the TextField.scss file and was requested to instead be made directly to the border token in the @shopify/polaris-tokens [repo](https://github.com/Shopify/polaris-tokens) (now part of this monorepo), so I'm _guessing_ that's why that PR was closed, but I'm not sure why we didn't follow through in updating the `--p-border-shadow` token.

We have some projects already underway looking at our color palette more holistically. We will certainly reference this issue during those projects as we find the best solution and do our best to make sure Polaris is meeting the needs of all our merchants.

**Comment by aardrian** [comment]
Thanks for the context.

Can you add the "Accesssibility" label to this issue?

What is the timeframe for the WCAG conformance work (or at least this SC)?

**Comment by heyjoethomas** [comment]
Label added ✅ 
Don't have any hard deadlines for you right now. Projects are still in semi-early design phase.

**Comment by yurm04** [comment]
Quick update: we're going to take a look at this and post a follow up tomorrow/next week as we weigh our options for how to move forward with a solution

**Comment by sarahill** [comment]
Update:

We explored some different options and have decided to move forward with increasing the contrast of the border color for the `TextField` component. Details attached in the screenshot below:

The contrast ratio is based off the top border of the text field against the background color

![text field contrast](https://user-images.githubusercontent.com/8629173/207162031-53b30d16-c3bd-4003-9402-df9addcc681d.png)

We're also increasing the contrast of the `Select` component to feel visually cohesive with the updated `TextField` since these are often displayed next to each other.

This only tackles a small portion of Polaris inputs but it heads us in the right direction. There are on-going projects that will look at color and the contrast of user interface elements as a whole. As Joe mentioned above, these are still in the early stages so we felt it was important to implement a solution now.

**Comment by aardrian** [comment]
I appreciate addressing this before the unknown future date for the overhaul.

Obviously I would prefer the _entire_ border is addressed, not just the top border (for reasons cited above), but I sense that will not be happening in the short term.

**Comment by sarahill** [comment]
I understand and agree addressing the whole border is a better solution. We felt this solution would allow us to make the change sooner and have immediate impact for merchants while we continue those explorations.

**Comment by sarahill** [comment]
Just circling back here with next steps:

- We merged the change to `TextField` in [this PR](https://github.com/Shopify/polaris/pull/7893)
- We'll be opening a new issue to revisit these elements to look as inputs as a whole

**Comment by aardrian** [comment]
@sarahill 
> - We'll be opening a new issue to revisit these elements to look as inputs as a whole

Can you please link that issue here when created?

**Comment by sarahill** [comment]
Created a follow up issue for this https://github.com/Shopify/polaris/issues/7977
